libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -430,7 +445,20 @@ bool ControlFlowBuilder::visit(Identifier const& _identifier)
 	return true;
 }
 
+bool ControlFlowBuilder::visitNode(ASTNode const& _node)
+{
+	solAssert(!!m_currentNode, "");
 
+	if (!m_currentNode->location.source)
```

Comment:
Can you explain why this function does not just reset the current location in all cases?

Reply:
A block of control flow has location that can be distinct from any AST node. It's always a continuous region, though, so this function successively extends the location for the control flow node to the whole area in the source that is covered by all AST nodes contained in the control flow block.

Reply:
Actually coming up with a nice way to figure out which source locations to report turns out to be much harder than actually detecting the unreachable code :-). Let me know if you see a problem with the current approach or have a better one.

Reply:
Offline discussion: This gathers all source locations that contribute to the CFG node.

Reply:
Please extract this functionality into a function of SourceLocation.

Reply:
Done.